src/readBuffer.js

```diff
@@ -1,12 +1,32 @@
+class EarlyEOF extends Error {
```

Comment:
I would avoid extending built-ins especially when Babel is used, because the transpiled code will not properly extend the built-in, and you need to write hacky code to work around it (setPrototypeOf has a big perf penalty)

I would recommend to just do something like Object.assign(new Error(‘Stream ended ...’), { name: ‘EarlyEofError’ }) and check for err.name instead of instanceof.

Reply:
See how its done in the [webapp with `createAggregateError`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/src/util/errors.ts#L36-40)

Reply:
Done